src/main/java/com/google/devtools/build/lib/remote/Chunker.java

```diff
@@ -142,13 +142,12 @@ public void reset() throws IOException {
    * <p>Closes any open resources (file handles, ...).
    */
   public void seek(long toOffset) throws IOException {
-    maybeInitialize();
+    boolean resetAndSkip = toOffset < offset;
     if (toOffset < offset) {
       reset();
-      if (toOffset != 0) {
-        data.skip(toOffset);
```

Comment:
if you added a maybeInitialize() above this line above data.skip() wouldn't that be enough?

Reply:
Yeah, there's probably too much cleverness going on in this already - I was avoiding the initialize and skip if it was a noop or could be left in the reset condition.